src/client/Microsoft.Identity.Client/Internal/Requests/SilentRequest.cs

```diff
@@ -29,294 +30,56 @@ internal class SilentRequest : RequestBase
             : base(serviceBundle, authenticationRequestParameters, silentParameters)
         {
             _silentParameters = silentParameters;
-        }
-
-        private bool BrokerIsInstalledAndSupportsSilentAuth
-        {
-            get
-            {
-                if (_canExecuteBrokerValue == null)
-                {
-                    _canExecuteBrokerValue = AuthenticationRequestParameters.IsBrokerConfigured && ServiceBundle.PlatformProxy.CanBrokerSupportSilentAuth();
-                }
-
-                return (bool)_canExecuteBrokerValue;
-            }
-        }
-
-        private async Task<IAccount> GetSingleAccountForLoginHintAsync(string loginHint)
-        {
-            var accounts = await CacheManager.GetAccountsAsync(ServiceBundle.Config.AuthorityInfo.CanonicalAuthority)
-                .ConfigureAwait(false);
-
-            accounts = accounts
-                .Where(a => !string.IsNullOrWhiteSpace(a.Username) &&
-                       a.Username.Equals(loginHint, StringComparison.OrdinalIgnoreCase))
-                .ToList();
-
-            if (!accounts.Any())
-            {
-                throw new MsalUiRequiredException(
-                    MsalError.NoAccountForLoginHint,
-                    MsalErrorMessage.NoAccountForLoginHint,
-                    null,
-                    UiRequiredExceptionClassification.AcquireTokenSilentFailed);
-            }
 
-            if (accounts.Count() > 1)
-            {
-                throw new MsalUiRequiredException(
-                    MsalError.MultipleAccountsForLoginHint,
-                    MsalErrorMessage.MultipleAccountsForLoginHint,
-                    null,
-                    UiRequiredExceptionClassification.AcquireTokenSilentFailed);
+            _brokerStrategy = new Lazy<ISilentAuthStrategy>(() => new SilentBrokerAuthStretegy(this, 
+                                                                                               serviceBundle, 
+                                                                                               authenticationRequestParameters, 
+                                                                                               silentParameters, 
+                                                                                               serviceBundle.PlatformProxy.CreateBroker(null)));
+            _clientStrategy = new SilentClientAuthStretegy(this, serviceBundle, authenticationRequestParameters, silentParameters);
 
-            }
-
-            return accounts.First();
+            _logger = authenticationRequestParameters.RequestContext.Logger;
         }
 
-        private async Task<IAccount> GetAccountFromParamsOrLoginHintAsync(AcquireTokenSilentParameters silentParameters)
+        //For test
+        internal void SetStrategiesForTest(ISilentAuthStrategy clientStrategy, ISilentAuthStrategy brokerStrategy)
```

Comment:
This is good, but the recommended pattern is to pass test implementations from the ctor, smth like 

```csharp
public SilentRequest(exisisting_params, ISilentAuthStrategy clientStrategyOverride, ISilentAuthStrategy brokerStrategyOverride)
{
_clientStrategy = clientStrategyOverride ?? new SilentClientAuthStretegy(...);
}
```

Reply:
The advantage of this is that you can keep the class immutable, i.e. _clientStrategy  and _brokerStrategy variables will be readonly and once the SilentRequest object is created, it cannot be changed. Immutable classes are a dream - they are simpler to work with and are thread safe.